vissl/data/ssl_dataset.py

```diff
@@ -118,6 +118,16 @@ def _get_data_limit_sampling(cfg: AttrDict, split: str) -> AttrDict:
         )
         return cfg["DATA"][split].get("DATA_LIMIT_SAMPLING", default_sampling)
 
+    def set_epoch(self, epoch):
+        for data_obj in self.data_objs:
+            if hasattr(data_obj, "set_epoch"):
+                data_obj.set_epoch(epoch)
```

Comment:
I think this will fail here if the data_obj is not airstore object? since we don't have the set_epoch implemented for other datasets.

Reply:
It has "if hasattr(data_obj, "set_epoch"):" check. I am thinking make the set_epoch an optional interface on dataset by duck typing. It make sense for all future iterator based dataset implementations, where they can not rely on sampler to do the checkpoint recovering. If we don't plan to have another one, we can hardcode it.

Reply:
I change the code following the suggestions, now it directly checking the dataset type